### ISSUE TITLE
Way to differentiate in CMake whether rez is building, build-installing, releasing

### ISSUE THREAD

**Issue by maxnbk** [nerdvegas/rez#305]
Some sort of environment-variable for differentiating whether rez is being run as rez-build, rez-build -i, rez-release, and perhaps rez-build -c -i, and/or possibly to hook on to the --prefix if specified, would be very useful to have access to from CMake's point of view when building a package for custom behavior when needed.

**Comment by bfloch** [comment]
In CMake you have something:
https://github.com/nerdvegas/rez/blob/master/src/rezplugins/build_system/cmake.py#L125

If you use your own build system you need to properly parse it.
I agree that it could be an environment variable per se but there has been discussion here:
https://github.com/nerdvegas/rez/issues/102

**Comment by maxnbk** [comment]
No way to tell if releasing, unfortunately.

**Comment by nerdvegas** [comment]
Yeah I'd like to clean this up - have rez provide a standard set of
env-vars that doesn't differ between build systems.

Thx
A

**Comment by fnaum** [comment]
Hi @maxnbk , 
The variable `CENTRAL` gets defined when doing a release, so you can use the following on your cmake

```
if(CENTRAL)
   do your stuff 
endif(CENTRAL)
```

we use that for example to only build the documentation at release time
